cartography/intel/aws/kms.py

```diff
@@ -0,0 +1,333 @@
+    # policy is a very complex format, so the policyuniverse library will be used for parsing out important data
+    # ...metadata...
+    # "Policy" :
+    # {
+    #   "Version": "2012-10-17",
+    #   "Id": "key-consolepolicy-5",
+    #   "Statement": [
+    #     {
+    #       "Sid": "Enable IAM User Permissions",
+    #       "Effect": "Allow",
+    #       "Principal": {
+    #         "AWS": "arn:aws:iam::123456789012:root"
+    #       },
+    #       "Action": "kms:*",
+    #       "Resource": "*"
+    #     },
+    #     {
+    #       "Sid": "Allow access for Key Administrators",
+    #       "Effect": "Allow",
+    #       "Principal": {
+    #         "AWS": "arn:aws:iam::123456789012:role/ec2-manager"
+    #       },
+    #       "Action": [
+    #         "kms:Create*",
+    #         "kms:Describe*",
+    #         "kms:Enable*",
+    #         "kms:List*",
+    #         "kms:Put*",
+    #         "kms:Update*",
+    #         "kms:Revoke*",
+    #         "kms:Disable*",
+    #         "kms:Get*",
+    #         "kms:Delete*",
+    #         "kms:ScheduleKeyDeletion",
+    #         "kms:CancelKeyDeletion"
+    #       ],
+    #       "Resource": "*"
+    #     }
+    #   ]
+    # }
+    if policy is not None:
+        # get just the policy element and convert to JSON because boto3 returns this as string
+        policy = Policy(json.loads(policy['Policy']))
+        if policy.is_internet_accessible():
+            return {
+                "kms_key": key,
+                "internet_accessible": True,
+                "accessible_actions": list(policy.internet_accessible_actions()),
+            }
+        else:
+            return None
+    else:
+        return None
+
+
+@timeit
+def load_kms_keys(neo4j_session, data, region, current_aws_account_id, aws_update_tag):
+    ingest_keys = """
+    UNWIND {key_list} AS k
+    MERGE (kmskey:KMSKey{id:k.KeyId})
+    ON CREATE SET kmskey.firstseen = timestamp(),
+    kmskey.arn = k.Arn, kmskey.creationdate = k.CreationDate
+    SET kmskey.deletiondate = k.DeletionDate,
+    kmskey.validto = k.ValidTo,
+    kmskey.enabled = k.Enabled,
+    kmskey.keystate = k.KeyState,
+    kmskey.customkeystoreid = k.CustomKeyStoreId,
+    kmskey.cloudhsmclusterid = k.CloudHsmClusterId,
+    kmskey.lastupdated = {aws_update_tag},
+    kmskey.region = {Region}
+    WITH kmskey
+    MATCH (aa:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MERGE (aa)-[r:RESOURCE]->(kmskey)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}
+    """
+
+    # neo4j does not accept datetime objects and values. This loop is used to convert
+    # these values to string.
+    for key in data:
+        key['CreationDate'] = str(key['CreationDate'])
+        key['DeletionDate'] = str(key.get('DeletionDate'))
+        key['ValidTo'] = str(key.get('ValidTo'))
+
+    # The owner data returned by the API maps to the aws account nickname and not the IAM user
```

Comment:
I see that this comment was copied from the s3 sync, is it still relevant here?

Reply:
No it's not. Removed it in latest commit :)

Reply:
I think you need to update your branch with the latest master and then I can merge it in.

Reply:
@achantavy  Done :D